src/pretix/base/models/event.py

```diff
@@ -191,7 +192,12 @@ def get_date_range_display(self, tz=None, force_show_end=False, as_html=False, t
                 (dt.astimezone(tz).date() == df.astimezone(tz).date() + timedelta(days=1) and
                  dt.astimezone(tz).time() < df.astimezone(tz).time())
             )
-        d = daterange(df.astimezone(tz), dt.astimezone(tz), as_html)
+
+        request_lang = get_language()
+        if '-' not in request_lang and self.settings.region:
+            request_lang += '-' + self.settings.region.lower()
+        with language(request_lang):
+            d = daterange(df.astimezone(tz), dt.astimezone(tz), as_html)
```

Comment:
Hmm, I don't really think get_date_range_display should need to call with language(). The idea of the context manager is not to call it on the lowest possible levels. I think we should figure out why it isn't called in the context where get_range_display is called and where the language is activated and fix it there.